Wrappers/Python/cil/optimisation/functions/KullbackLeibler.py

```diff
@@ -1,31 +1,239 @@
-# -*- coding: utf-8 -*-
-#   This work is part of the Core Imaging Library (CIL) developed by CCPi 
-#   (Collaborative Computational Project in Tomographic Imaging), with 
-#   substantial contributions by UKRI-STFC and University of Manchester.
+# Copyright 2022 United Kingdom Research and Innovation
+# Copyright 2022 The University of Manchester
 
-#   Licensed under the Apache License, Version 2.0 (the "License");
-#   you may not use this file except in compliance with the License.
-#   You may obtain a copy of the License at
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
 
-#   http://www.apache.org/licenses/LICENSE-2.0
+#     http://www.apache.org/licenses/LICENSE-2.0
 
-#   Unless required by applicable law or agreed to in writing, software
-#   distributed under the License is distributed on an "AS IS" BASIS,
-#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-#   See the License for the specific language governing permissions and
-#   limitations under the License.
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
 
 import numpy
 from cil.optimisation.functions import Function
 from numbers import Number
-import functools
 import scipy.special
+import logging
+
 try:
     import numba
     from numba import jit, prange
     has_numba = True
-    '''Some parallelisation of KL calls'''
+except ImportError as ie:
+    has_numba = False
+
+class KullbackLeibler(Function):
+
+    r""" Kullback Leibler
+            
+    .. math:: F(u, v)
+            = \begin{cases} 
+            u \log(\frac{u}{v}) - u + v & \mbox{ if } u > 0, v > 0\\
+            v & \mbox{ if } u = 0, v \ge 0 \\
+            \infty, & \mbox{otherwise}
+            \end{cases}  
+
+    where the :math:`0\log0 := 0` convention is used.  
+              
+    **kwargs
+        b : DataContainer, non-negative      
+            Translates the function at point :code:`b`.
+        mask : DataContainer, default = None
+               Mask for the data :code:`b`
+        eta : DataContainer, default = array of zeros
+              Background noise
+
+    Raises
+    ------
+    ValueError
```

Comment:
I think this section is unnecessary and bulks the docsting. We do not use `Raises` elsewhere, and if it raises an error the user will soon find out,